starport/interface/cli/starport/cmd/network_proposal_approve.go

```diff
@@ -42,12 +47,36 @@ func networkProposalApproveHandler(cmd *cobra.Command, args []string) error {
 		proposalList = args[1]
 	)
 
-	nb, err := newNetworkBuilder()
+	ev := events.NewBus()
+	go printEvents(ev, s)
+
+	// Temporary home to test proposals
+	tmpHome, err := os.MkdirTemp("", "")
 	if err != nil {
 		return err
 	}
+	defer os.RemoveAll(tmpHome)
 
-	var reviewals []spn.Reviewal
+	// Initialize the blockchain
+	nb, err := newNetworkBuilder(networkbuilder.CollectEvents(ev))
+	if err != nil {
+		return err
+	}
+
+	blockchain, err := nb.Init(
```

Comment:
I see that this one is used in three ways, can we wrap the similar logic in a func?

Reply:
We can actually have the temporary dir creation in `VerifyProposal` function. For show chain command I think it's good since we should create and remove the dir inside the same function